src/conversion.rs

```diff
@@ -34,156 +37,286 @@ impl fmt::Display for Error {
     }
 }
 
-fn create_geo_coordinate<T>(coord: &Coord<T>) -> geo_types::Coordinate<T>
+impl<T> TryFrom<Wkt<T>> for geo_types::Geometry<T>
 where
     T: num_traits::Float,
 {
-    geo_types::Coordinate {
-        x: T::from(coord.x).unwrap(),
-        y: T::from(coord.y).unwrap(),
+    type Error = Error;
+
+    fn try_from(mut wkt: Wkt<T>) -> Result<Self, Self::Error> {
+        if wkt.items.len() == 1 {
+            Self::try_from(wkt.items.pop().unwrap())
+        } else {
+            Geometry::GeometryCollection(GeometryCollection(wkt.items)).try_into()
```

Comment:
It seems like a WKT can have multiple Geometries in it - even outside of the context of a GeometryCollection.

If WKT is going to be convertible to a `geo_type::Geometry` we can't have that, so I've opted to wrap it in a collection if it's more than one.

Even though this `impl` is the whole reason I created this PR, I won't be offended if you'd prefer to omit this part. I think the other changes are an improvement even if this part gets dropped.